### ISSUE TITLE
[CLOSED] ext2: sync buffers if requested

### ISSUE THREAD

**Issue by zilinc** [NICTA/cogent#173]
<a href="https://github.csiro.au/ahixon"><img src="https://github.csiro.au/avatars/u/257?" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [ahixon](https://github.csiro.au/ahixon)**
_Tue Mar 24 14:23:46 2015_
_Originally opened as https://github.csiro.au/ts-filesystems/Cogent/issues/68_

----

Currently prevents us from unmounting correctly after writing to files, since we have unwritten buffers inside the inode. Caused from: `BUG_ON(!list_empty(&inode->i_data.private_list));`

Basically, if writeback control is set to `WC_ALL`, then we need to sync_dirty_buffer forcefully. This presumably gets called on all dirty inodes at mount time.

If this doesn't solve our problem, then... :sob:

**Comment by zilinc** [comment]
<a href="https://github.csiro.au/ahixon"><img src="https://github.csiro.au/avatars/u/257?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ahixon](https://github.csiro.au/ahixon)**
_Tue Mar 24 15:17:53 2015_

----

Sweet. Yup, 24b4a251316ad912f0c5596f1c8b87c250ba4970 fixed it.

**Comment by zilinc** [comment]
<a href="https://github.csiro.au/ahixon"><img src="https://github.csiro.au/avatars/u/257?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ahixon](https://github.csiro.au/ahixon)**
_Tue Mar 24 15:19:45 2015_

----

Or not... :-1:

**Comment by zilinc** [comment]
<a href="https://github.csiro.au/ahixon"><img src="https://github.csiro.au/avatars/u/257?" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ahixon](https://github.csiro.au/ahixon)**
_Tue Mar 24 20:59:48 2015_

----

Fixed in e8e4dd4a239a47a352aa7f4c2d8b0d57ac85daa5
